cmd/cosign/cli/generate_key_pair.go

```diff
@@ -105,3 +102,18 @@ func getPass(confirm bool) ([]byte, error) {
 	}
 	return pw1, nil
 }
+
+func setReadPasswordFn() {
+	if Read != nil {
+		return
+	}
+	// Handle piped in passwords.
+	Read = func() ([]byte, error) {
```

Comment:
should this get defined up top? I don't love having this sit around as nil. We can still override it in here if it's not a terminal

Reply:
yah that's better, fixed